Switch the CI from python 3.11-dev to 3.11

rokm opened the pull request:
Switch from python 3.11-dev to 3.11, which runs the full-blown CI with `requirements-libraries.txt`. 

The `PySide2` and `PySide6` wheels seem to explicitly require `python < 3.11`, so they need to go under version check. 

The ancient Django 2.1.8 also seems to run into some issue under 3.11; updating it to the latest version breaks the test, but that's because the test project that we use is outdated and uses functionality that is unavailable in Django 4.x. Instead of manually upgrading the test project, I've opted to generate a new one from scratch. ~For python 3.7, the latest available version is 3.2.16, and it passes the test even though the test project was generated using latest version.~ We do not test Django under python 3.7 anymore.

~Sync requirements.txt and setup.cfg; in particular, copy the minimum required versions from setup.cfg into requirements.txt~. Remove the `requirements.txt` file. And set the minimum required importlib-metadata version in `setup.cfg` to prevent people from somehow ending up with a really ancient version that is older than what is in python 3.8 stdlib.

rokm committed:
ci: switch python 3.11-dev to 3.11

rokm committed:
tests: requirements: restrict PySide2 and PySide6 to python < 3.11

Currently available wheels for PySide2 and PySide6 seems to
explicitly require python < 3.11, so we cannot use them.

rokm committed:
tests: django: update the test django project

Re-generate the test django project using
`django-admin startproject django_site`
under Django 4.1.2. Fixes the `ImportError: cannot import name
'url' from 'django.conf.urls'` error due to `url` being deprecated
in Django 3.0 and removed in 4.0.

Also run the migrations to populate the database.

rokm committed:
tests: requirements: repin Django to 4.1.2

Repin Django to 4.1.2 for python >= 3.8, and drop the test for
python 3.7 (where the last available version is 3.2.16).

rokm committed:
setup.cfg: set minimum version for importlib-metadata

On python < 3.8, require importlib-metadata >= 1.4, which is the
version that matches the stdlib version in python 3.8.

bwoodsend commented:
> Instead of manually upgrading the test project, I've opted to generate a new one from scratch.

But this was our golden excuse to delete that test for good...

bwoodsend commented on tests/requirements-libraries.txt:
```diff
@@ -11,21 +11,13 @@
 # -------
 # These packages work with no (known) issues.
 babel==2.10.3
+Django==3.2.16; python_version < '3.8'
```
I'd rather drop the 3.7 variant. If Django don't support 3.7 then I don't see why we should try to ensure Django+Python 3.7 compatibility.

rokm replied:
Fair enough. Will drop it in the next revision.

bwoodsend commented on requirements.txt:
```diff
@@ -8,9 +8,9 @@
+pefile >= 2022.5.30; sys_platform == 'win32'
+pywin32-ctypes >= 0.2.0; sys_platform == 'win32'
+macholib >= 1.8; sys_platform == 'darwin'
+importlib-metadata >= 1.4; python_version < '3.8'
```
Why do we still have this file?

rokm replied:
That's a good question. Shall we take this opportunity to get rid of it instead of trying to sync it up to setup.cfg?

bwoodsend replied:
I think so. Now that I think about it, it was probably there for when we had those `pyinstaller.py`/`pyi_makespec.py` files and running PyInstaller without installing it was considered supported.

rokm commented:
> > Instead of manually upgrading the test project, I've opted to generate a new one from scratch.
> 
> But this was our golden excuse to delete that test for good...

It would have been, had the test required more effort to fix...

rokm committed:
Remove requirements.txt

We do not really support running PyInstaller from source directory
without installing it anymore, so requirements.txt is now
redundant. Even development installs should be done using
"pip install -e ."
